src/core/externalstorage/qgshttpexternalstorage.cpp

```diff
@@ -175,31 +189,69 @@ void QgsWebDAVExternalStorageFetchedContent::onFetched()
   }
 }
 
-void QgsWebDAVExternalStorageFetchedContent::cancel()
+void QgsHttpExternalStorageFetchedContent::cancel()
 {
   mFetchedContent->cancel();
 }
 
-QString QgsWebDAVExternalStorage::type() const
+
+// WEB DAV PROTOCOL
+
+QString QgsWebDavExternalStorage::type() const
 {
   return QStringLiteral( "WebDAV" );
 };
 
-QString QgsWebDAVExternalStorage::displayName() const
+QString QgsWebDavExternalStorage::displayName() const
 {
   return QObject::tr( "WebDAV Storage" );
 };
 
-QgsExternalStorageStoredContent *QgsWebDAVExternalStorage::doStore( const QString &filePath, const QString &url, const QString &authcfg ) const
+QgsExternalStorageStoredContent *QgsWebDavExternalStorage::doStore( const QString &filePath, const QString &url, const QString &authcfg ) const
 {
-  return new QgsWebDAVExternalStorageStoredContent( filePath, url, authcfg );
+  return new QgsHttpExternalStorageStoredContent( filePath, url, authcfg );
 };
 
-QgsExternalStorageFetchedContent *QgsWebDAVExternalStorage::doFetch( const QString &url, const QString &authConfig ) const
+QgsExternalStorageFetchedContent *QgsWebDavExternalStorage::doFetch( const QString &url, const QString &authConfig ) const
 {
   QgsFetchedContent *fetchedContent = QgsApplication::networkContentFetcherRegistry()->fetch( url, Qgis::ActionStart::Deferred, authConfig );
 
-  return new QgsWebDAVExternalStorageFetchedContent( fetchedContent );
+  return new QgsHttpExternalStorageFetchedContent( fetchedContent );
 }
 
+
+// AWS S3 PROTOCOL
+
+QString QgsAwsS3ExternalStorage::type() const
+{
+  return QStringLiteral( "AWSS3" );
+};
+
+QString QgsAwsS3ExternalStorage::displayName() const
+{
+  return QObject::tr( "AWS S3" );
+};
+
+QgsExternalStorageStoredContent *QgsAwsS3ExternalStorage::doStore( const QString &filePath, const QString &url, const QString &authcfg ) const
+{
+  QgsHttpExternalStorageStoredContent *storedContent = new QgsHttpExternalStorageStoredContent( filePath, url, authcfg );
+  storedContent->setPrepareRequestHandler( []( QNetworkRequest & request, QFile * f )
+  {
+    QCryptographicHash payloadCrypto( QCryptographicHash::Sha256 );
+    payloadCrypto.addData( f );
+    QByteArray payloadHash = payloadCrypto.result().toHex();
+    f->seek( 0 );
+    request.setRawHeader( QByteArray( "X-Amz-Content-SHA256" ), payloadHash );
+    request.setAttribute( QNetworkRequest::User, f->fileName() );
```

Comment:
What is it used for ?

Reply:
My mistake, it should not be here (remaining from a test)